course3/homework-list/list.h

```diff
@@ -0,0 +1,105 @@
+#pragma once
+
+#include <memory>
+
+
+namespace Karun
+{
+
+template<typename T>
+class List
+{
+public:
+	class Node
+	{
+	friend class List;
+	
+	public:
+		Node(const T& data, std::unique_ptr<Node>& next)
+		: 	mData(data)
+		{
+			mNext = std::move(next);
+		}
+		
+	private:
+		T mData;
+		std::unique_ptr<Node> mNext;
+	};
+	
+	class iterator
+	{
+	friend class List;
+		
+	public:
+		iterator(Node* node) : mNode(node) {}
```

Comment:
Using raw pointers is absolutely fine in this case. The use of raw pointer makes the following clear:
* The pointer is not-owning (otherwise it would be a smart pointer)
* The pointer can be null (otherwise it would be a reference)

IMO this is one of few situations when raw pointer is the appropriate type.